doc/specs/Proto extensions-spec.md

```diff
@@ -0,0 +1,139 @@
+
+#### Modifications to existing profiles
+
+For modifications to existing profiles, the json stub would need to indicate which profile it wishes to modify.
+A straightforward way to do this would be by providing the name of the profile. Since this modification would
+occur at the step _before_ any user changes to names are layered, the names of the profiles are consistent and
+can be used for identification. For example, the name of a profile generated by the WSL profile generator is
+simply the argument to "wsl.exe -d \<name\>", the name of a profile generated by the Powershell profile generator
+would be "Powershell Core" or "Powershell (ARM)" etc depending on the version.
+
+We might run into the case where multiple json stubs modify the same profile and so they override each other. We
+will need some sort of hierarchy to determine which stub to ultimately use. My initial thought is that we could
+check the owner of the json file and see if it matches the profile they are trying to modify (so something like
+only Powershell can modify the Powershell profiles and only Ubuntu can modify the Ubuntu profile etc). Independent
+parties will need to create full profiles with the desired commandline argument instead.
+
+#### Full profile stubs
+
+Technically, full profile stubs do not need to contain anything (they could just be '\{\}'). However we should
+have some qualifying minimum criteria before we accept a stub as a full profile. I suggest that we only create
+new profile objects from stubs that contain at least the following
+
+* A name (and this name _should not_ match the names of existing profiles as described above, because that would cause us to interpret this profile as modifying an existing one)
+* A commandline argument
+* A unique GUID - this is so we avoid conflicts if several different creators name their profile the same thing
+
+As in the case of the dynamic profile generator, if we create a profile that did not exist before (i.e. does not
+exist in the user settings), we need to add the profile to the user settings file and re-save that file. 
+
+#### Colour schemes
+
+Once again, we could run into the problem of several json stubs wanting to modify the same colour scheme. We
+will require a hierarchy here as well. Though since the 'owner' of a colour scheme is a lot more dubious, I wonder
+if we should only allow creation of new colour schemes for now and ignore requests to modify existing colour
+schemes. 
+
+### Location(s) of the json files
+
+For apps that are installed 'traditionally', their installer could simply add a json file to a specific local folder
+(specified by us) that we will look through every time we generate profiles. This is also where users can add
+their own json files for Terminal to generate profiles from. 
+
+For apps that are installed through something like the Microsoft Store, they should add their json stub(s) to
+an app extension. We will then obtain them during profile generation. 

```

Comment:
What does this look like? How do we detect the app extension (or is it their job to update us?)